tracer/span.go

```diff
@@ -121,12 +141,20 @@ func (s *Span) SetMetric(key string, val float64) {
 		return
 	}
 
-	s.mu.Lock()
+	s.Lock()
+	defer s.Unlock()
+
+	// We don't lock spans when flushing, so we could have a data race when
+	// modifying a span as it's being flushed. This protects us against that
+	// race, since spans are marked `finished` before we flush them.
```

Comment:
What I'm trying to say with this comment is that even though we've locked the span here, it still isn't safe to modify `s.Metrics` since the flushing code doesn't lock the spans. I can update these comments to be a bit clearer.